Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +58,181 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
+        #region Server
+
+        private async Task<ListLocationsResponse> ListLocations(ListLocationsRequest request)
+        {
+            return await _amazonGameLiftClient.ListLocationsAsync(request);
+        }
+
+        private async Task<CreateLocationResponse> CreateLocation(CreateLocationRequest request)
+        {
+            return await _amazonGameLiftClient.CreateLocationAsync(request);
+        }
+
+        private async Task<RegisterComputeResponse> RegisterCompute(RegisterComputeRequest request)
         {
-            return new AmazonGameLiftClient(s_dummyAwsAccessKey, s_dummyAwsSecretKey, new AmazonGameLiftConfig
+            return await _amazonGameLiftClient.RegisterComputeAsync(request);
+        }
+
+        private async Task<GetComputeAuthTokenResponse> GetComputeAuthToken(GetComputeAuthTokenRequest request)
+        {
+            return await _amazonGameLiftClient.GetComputeAuthTokenAsync(request);
+        }
+
+        private Task<CreateFleetResponse> CreateFleet(CreateFleetRequest request)
+        {
+            return _amazonGameLiftClient.CreateFleetAsync(request);
+        }
+        
+        private RetriveAwsCredentialsResponse RetrieveAwsCredentials(string profileName)
+        {
+            var request = new RetriveAwsCredentialsRequest { ProfileName = profileName };
+            return _credentialsStore.RetriveAwsCredentials(request);
+        }
+        
+        public async Task UpdateAuthToken()
+        {
+            try
+            {
+                var config = _gameLiftClientSettings.GetGameLiftAnywhereConfiguration();
+                await GenerateAuthToken(config.ComputeName, config.FleetID);
+            }
+            catch (Exception e)
             {
-                ServiceURL = localEndpoint
-            });
+                Console.WriteLine(e);
+                throw;
+            }
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        private RetriveAwsCredentialsResponse SetupCredentials(string profileName)
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            var config = _gameLiftClientSettings.GetGameLiftAnywhereConfiguration();
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+            var credentials = RetrieveAwsCredentials(profileName);
+            return credentials;
+        }
+
+        private async Task GenerateAuthToken(string computeName, string fleetId)
+        {
+            try
+            {
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                _gameLiftClientSettings.authToken = computeAuthTokenResponse.AuthToken;
```

Comment:
Similar to the above comment, this method should only return the AuthToken, and the invoker should be responsible for updating the _gameLiftClientSettings where required. The Wrapper should only include calls to the sdk and nothing more imo. Happy to discuss if you feel otherwise

Reply:
Agree, will move this to UI when available.